sky/clouds/azure.py

```diff
@@ -302,21 +305,27 @@ def failover_disk_tier(
                 specified configuration is not a valid combination, and should
                 not be used for launching a VM.
             """
-            if disk_tier is not None:
+            if (disk_tier is not None and
+                    disk_tier != resources_utils.DiskTier.BEST):
                 ok, _ = Azure.check_disk_tier(instance_type, disk_tier)
                 return (True, disk_tier) if ok else (False, None)
-            disk_tier = clouds.Cloud._DEFAULT_DISK_TIER
-            all_tiers = {'high', 'medium', 'low'}
-            while not Azure.check_disk_tier(instance_type, disk_tier)[0]:
-                all_tiers.remove(disk_tier)
-                if not all_tiers:
-                    # No available disk_tier found the specified instance_type
-                    return (False, None)
-                disk_tier = list(all_tiers)[0]
-            if disk_tier != clouds.Cloud._DEFAULT_DISK_TIER:
-                return True, disk_tier
+            # All tiers in reversed order, i.e. from best to worst.
+            # We will failover along this order.
+            all_tiers = list(reversed(resources_utils.DiskTier))
+            if disk_tier is None:
+                # Start failover from the default disk tier.
+                start_index = all_tiers.index(clouds.Cloud._DEFAULT_DISK_TIER)
             else:
-                return True, None
+                # Start failover from the best disk tier.
+                start_index = all_tiers.index(
+                    Azure.translate_disk_tier(disk_tier))
```

Comment:
If we start with best tier, would the following always be `True`? How does the failover work then, if it does not fill in the actual disk_tier needed? It seems the later logic will not do failover for `BEST` disk_tier.

Reply:
To discuss: I'm thinking maybe we could disable the auto-failover for all non-best tiers in Azure? The current behavior is basically the same in Azure...